megatron/model/language_model.py

```diff
@@ -150,7 +151,9 @@ def __init__(self,
                 max_sequence_length, self.hidden_size)
             self._position_embeddings_key = 'position_embeddings'
             # Initialize the position embeddings.
-            self.init_method(self.position_embeddings.weight)
+            with deepspeed.zero.GatheredParameters(self.position_embeddings.weight, modifier_rank=0):
```

Comment:
does this context manager interfere with things if we don't have ZeRO-3 on? do we need to add a flag?

Reply:
Nope! It doesn’t interfere with things, even if we have ZeRO completely off